go_build_manifest/build.go

```diff
@@ -97,70 +97,83 @@ var totalImports = 0
+func transformVersion(inVersion string) string {
+	var outVersion string = strings.Replace(inVersion, "v", "", 1)
+
+	return strings.Split(outVersion, "+")[0]
 }
 
+// processGraphData ... Executes go graph command and process it to get direct dependencies.
 func processGraphData() int {
 	// Get graph data
-	cmdGoModGraph := exec.Command("go", "mod", "graph")
+	cmdGoModGraph := executeCommand("go", "mod", "graph")
 	cmdGoModGraph.Dir = sourceRootFolder
-	if output, err := cmdGoModGraph.Output(); err != nil {
+	output, err := cmdGoModGraph.Output()
+
+	if err != nil {
 		fmt.Printf("ERROR :: Command `go mod graph` failed, resolve project build errors. %s\n", err)
 		return -1
-	} else {
-		for _, value := range strings.Split(string(output), "\n") {
-			if len(value) > 0 {
-				// Extract direct dependency from go.mod
-				pc := strings.Split(value, " ")
-				if !strings.Contains(pc[0], "@") {
-					mainModule = pc[0]
-					mv := strings.Split(pc[1], "@")
-					directDependencies[mv[0]] = DirectDependency{mv[0], mv[1], false, make([]DirectPackage, 0), make([]Transitive, 0)}
-					totalDirectModuleDependencies++
-				}
+	}
+
+	for _, value := range strings.Split(string(output), "\n") {
+		if len(value) > 0 {
+			// Extract direct dependency from go.mod
+			pc := strings.Split(value, " ")
+			if !strings.Contains(pc[0], "@") {
+				mainModule = pc[0]
+				mv := strings.Split(pc[1], "@")
+				directDependencies[mv[0]] = DirectDependency{mv[0], mv[1], false, make([]DirectPackage, 0), make([]Transitive, 0)}
+				totalDirectModuleDependencies++
 			}
 		}
-		fmt.Println("Direct module dependencies:", totalDirectModuleDependencies)
 	}
+	fmt.Println("Direct module dependencies:", totalDirectModuleDependencies)
 
 	return 0
 }
 
+// processDepsData ... Get deps data through go list deps command and converts json into objects.
 func processDepsData() int {
 	// Switch to code directory and get graph
-	cmdGoListDeps := exec.Command("go", "list", "-json", "-deps", "./...")
+	cmdGoListDeps := executeCommand("go", "list", "-json", "-deps", "./...")
 	cmdGoListDeps.Dir = sourceRootFolder
-	if output, err := cmdGoListDeps.Output(); err != nil {
+	output, err := cmdGoListDeps.Output()
+
+	if err != nil {
 		fmt.Println("ERROR :: Command `go list -json -deps ./...` failed, resolve project build errors.", err)
 		return -1
-	} else {
-		goListDepsData := string(output)
-		goListDepsData = "{\"Packages\": [" + strings.ReplaceAll(goListDepsData, "}\n{", "},\n{") + "]}"
-
-		json.Unmarshal([]byte(goListDepsData), &goPackages)
-		totalDependencyPackages = len(goPackages.Packages)
-		fmt.Println("Packages in deps:", totalDependencyPackages)
 	}
+
+	goListDepsData := string(output)
+	goListDepsData = "{\"Packages\": [" + strings.ReplaceAll(goListDepsData, "}\n{", "},\n{") + "]}"
+
+	json.Unmarshal([]byte(goListDepsData), &goPackages)
+	totalDependencyPackages = len(goPackages.Packages)
+	fmt.Println("Packages in deps:", totalDependencyPackages)
```

Comment:
Use some logger, like zerolog.

Reply:
Basic log package is used to ouput the logs.

Reply:
Integrated zerolog now with context.